src/Tools/DynamoShapeManager/Preloader.cs

```diff
@@ -88,7 +88,7 @@ public Preloader(string rootFolder, IEnumerable<LibraryVersion> versions)
             var versionList = versions.ToList();
```

Comment:
Hi Ian, few things:

- Please add the following documentation:

```csharp
/// <param name="rootFolder">This method makes use of DynamoInstallDetective
/// to determine the installation location of various Autodesk products. This 
/// argument is not optional and must represent the full path to the folder 
/// which contains DynamoInstallDetective.dll. An exception is thrown if the 
/// assembly cannot be located.</param>
```

- Please add the null check for public API:

```csharp
if (string.IsNullOrEmpty(rootFolder))
    throw new ArgumentNullException("rootFolder");
if (!Directory.Exists(rootFolder))
    throw new DirectoryNotFoundException(rootFolder);
```

- Please add one more file check in `GetAsmInstallations` method below:

```csharp
var assembly = Assembly.LoadFrom(Path.Combine(rootFolder, "DynamoInstallDetective.dll"));
if (!File.Exists(assembly))
    throw new FileNotFoundException(assembly);
```

Then after that, LGTM. Thanks!